src/API/dashboard/dashboard.test.js

```diff
@@ -44,11 +47,20 @@ describe('## Dashboard APIs', () => {
             request(app)
                 .get('/api/dashboard')
                 .then(res => {
-                    expect(res.body.devices.length).equals(1);
                     expect(res.body.devices[0].device.mqttId).equals(device.mqttId);
                     done();
                 });
         });
+
+        it('should filter unpopulated sensors', done => {
+            request(app)
+                .get('/api/dashboard')
+                .expect(httpStatus.OK)
+                .then(res => {
+                    expect(res.body.devices.length).equals(1);
```

Comment:
It is better to set result to const here and expect it below, out of "then" callback